v2/api/appconfiguration/customizations/configuration_store_extensions.go

```diff
@@ -0,0 +1,148 @@
+
+	hasSecrets := secretsSpecified(typedObj)
+	if !hasSecrets {
+		log.V(Debug).Info("No secrets retrieval to perform as operatorSpec is empty")
+		return nil, nil
+	}
+
+	id, err := genruntime.GetAndParseResourceID(obj)
+	if err != nil {
+		return nil, err
+	}
+
+	keys := make(map[string]armappconfiguration.APIKey)
+	// Only bother calling ListKeys if there are secrets to retrieve
+	if hasSecrets {
+		subscription := armClient.SubscriptionID()
+		// Using armClient.ClientOptions() here ensures we share the same HTTP connection, so this is not opening a new
+		// connection each time through
+		var confClient *armappconfiguration.ConfigurationStoresClient
+		confClient, err = armappconfiguration.NewConfigurationStoresClient(subscription, armClient.Creds(), armClient.ClientOptions())
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to create new ConfigurationStoresClient")
+		}
+
+		var pager *runtime.Pager[armappconfiguration.ConfigurationStoresClientListKeysResponse]
+		var resp armappconfiguration.ConfigurationStoresClientListKeysResponse
+		pager = confClient.NewListKeysPager(id.ResourceGroupName, obj.AzureName(), nil)
+		for pager.More() {
+			resp, err = pager.NextPage(ctx)
+			secretsByName(resp.Value, keys)
+
+		}
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to retreive response")
+		}
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed listing keys")
+		}
+
+	}
+
+	secretSlice, err := secretsToWrite(typedObj, keys)
+	if err != nil {
+		return nil, err
+	}
+
+	return secretSlice, nil
+}
+
+func secretsSpecified(obj *storage.ConfigurationStore) bool {
+	if obj.Spec.OperatorSpec == nil || obj.Spec.OperatorSpec.Secrets == nil {
+		return false
+	}
+
+	secrets := obj.Spec.OperatorSpec.Secrets
+
+	if secrets.PrimaryKeyID != nil ||
+		secrets.SecondaryKeyID != nil ||
+		secrets.PrimaryReadOnlyKeyID != nil ||
+		secrets.SecondaryReadOnlyKeyID != nil ||
+		secrets.PrimaryKey != nil ||
+		secrets.SecondaryKey != nil ||
+		secrets.PrimaryReadOnlyKey != nil ||
+		secrets.SecondaryReadOnlyKey != nil ||
+		secrets.PrimaryConnectionString != nil ||
+		secrets.SecondaryConnectionString != nil ||
+		secrets.PrimaryReadOnlyConnectionString != nil ||
+		secrets.SecondaryReadOnlyConnectionString != nil {
+		return true
+	}
+
+	return false
+}
+
+func secretsByName(keys []*armappconfiguration.APIKey, result map[string]armappconfiguration.APIKey) {
+	for _, key := range keys {
+		if key == nil || key.Name == nil {
+			continue
+		}
+		result[*key.Name] = *key
+	}
+}
+
+func secretsToWrite(obj *storage.ConfigurationStore, keys map[string]armappconfiguration.APIKey) ([]*v1.Secret, error) {
+	operatorSpecSecrets := obj.Spec.OperatorSpec.Secrets
+	if operatorSpecSecrets == nil {
+		return nil, errors.Errorf("unexpected nil operatorspec")
+	}
+
+	collector := secrets.NewSecretCollector(obj.Namespace)
+	collector.AddSecretValue(operatorSpecSecrets.PrimaryConnectionString, *keys["Primary"].ConnectionString)
```

Comment:
Should we protect against if `keys` doesn't have these values?

Maybe add some if-guards and return error rather than here where we would panic? Could read them into vars too to avoid repeating the same `*keys["Primary"]` over and over too.

Reply:
Done!